### ISSUE TITLE
Update the syntax documentation examples output

### ISSUE THREAD

**Issue by sir4ur0n** [tweag/nickel#887]
**Describe the bug**
It seems that the output of some Nickel commands was improved since the documentation was written.
It would probably be good to reflect this in the documentation.

**To Reproduce**
E.g.
```
> nickel repl
nickel> let n = 5 in "The number %{n}."
error: type error
  ┌─ repl-input-12:1:14
  │
1 │ let n = 5 in "The number %{n}."
  │         -    ^^^^^^^^^^^^^^^^^^ this expression has type Num, but String was expected
  │         │
  │         evaluated to this
  │
  = interpolated string
```
while in the [documentation](https://nickel-lang.org/user-manual/syntax) it says:
```
nickel> let n = 5 in "The number %{n}."
error: Type error
```

The reality is much better than the documentation lets the reader think, this is a missed opportunity :smile: 

**Expected behavior**
Documentation is in line with reality

**Environment**
 - Version of the code: `master`

High-level question:
This issue can be solved "quickly" by just updating all the documentation example output. However such discrepancies are doomed to happen again in the future.

A longer but more future-proof alternative could be to dynamically generate the outputs by running `nickel` in the CI on those examples. This is future proof at the expense of initial implementation and probably a slightly more complex CI.

**Comment by sir4ur0n** [comment]
@bew I fully agree that the error message is quite confusing :+1: Nice catch!

Would you mind opening another issue about that please? :pray: To keep concerns separated

**Comment by yannham** [comment]
I think this is solved by #1611, which introduced automatic testing of all the manual snippets (including the output of REPL sessions), which now catch such mismatches as part of the CI.
